### PR TITLE
feat(editable-title): move cursor and scroll to the end

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

This PR improves the UX of editable title component.  Move the cursor and scroll to the end when focusing, for easier delete.

#### before
https://user-images.githubusercontent.com/67837651/121760645-b39fa000-cae0-11eb-96c2-c64010f4584e.mov
#### after

https://user-images.githubusercontent.com/11830681/122640420-fd683600-d131-11eb-8190-c26b0b1658cb.mov

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/15130#issuecomment-859975339
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/EditableTitle/index.tsx [hunk 1]
```diff
@@ -70,6 +70,13 @@ export default function EditableTitle({
   useEffect(() => {
     if (isEditing) {
       contentRef.current.focus();
+      // move cursor and scroll to the end
+      if (contentRef.current.setSelectionRange) {
+        const { length } = contentRef.current.value;
+        contentRef.current.setSelectionRange(length, length);
+        contentRef.current.scrollLeft = contentRef.current.scrollWidth;
+        contentRef.current.scrollTop = contentRef.current.scrollHeight;
```
Comment (villebro) [inline superset-frontend/src/components/EditableTitle/index.tsx:78]: Not sure if this is needed, but it probably doesn't hurt, either

**Review by villebro — APPROVED** [review]
LGTM, thanks for the fix!

File: superset-frontend/src/components/EditableTitle/index.tsx (hunk 1 above)
Comment (stephenLYZ) [reply superset-frontend/src/components/EditableTitle/index.tsx:78]: Cool. Here I think it need to scroll to the bottom for the textarea.

Comment (villebro) [reply superset-frontend/src/components/EditableTitle/index.tsx:78]: > Cool. Here I think it need to scroll to the bottom for the textarea.

Right you are! 👍

**Pull request merged**